client/api_client.go

```diff
@@ -34,7 +37,8 @@ func NewAPIClient(c Config) *APIClient {
 		httpClient = &http.Client{Transport: tr}
 	}
 
-	doer := newDebugRequestDoer(httpClient)
+	doer := wrapDebugRequestDoer(httpClient)
+	doer = wrapRetryRequestDoer(doer)
```

Comment:
Hmm this looks like it's overwrite the debug doer at first glance. Maybe simplify this. Also it's a bit confusing that the function names are the same but one takes a client and the other takes a doer.

## Option 1
```
debugDoer := wrapDebugRequestDoer(httpClient)
doer := wrapRetryRequestDoer(debugDoer)
```

## Option 2
```
doer := wrapRetryRequestDoer(wrapDebugRequestDoer(httpClient))
```

## Option 3
```
debugDoer := wrapDebugRequestDoer(httpClient)
rertyDoer := wrapRetryRequestDoer(httpClient)
doer := newDoer(doer ...rclient.RequestDoer)
```

Reply:
Ok, I'll go w/ option 1